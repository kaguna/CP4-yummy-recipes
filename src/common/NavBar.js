import React, { Component } from 'react';
import ChangePassword from '../components/users/ChangePassword';

class Header extends Component {
  render() {
    if (!localStorage.token) {
      window.location.assign('/login');
    }
    return (
      <nav className="navbar navbar-default">
        <ChangePassword />
        <div className="container-fluid">
          <div className="navbar-header">
            <button type="button" className="navbar-toggle" data-toggle="collapse" data-target="#myNavbar">
              <span className="icon-bar" />
              <span className="icon-bar" />
              <span className="icon-bar" />
            </button>
            <a className="navbar-brand" href=""><i className="glyphicon glyphicon-cutlery" /> Yummy Recipes</a>
          </div>
          <div className="collapse navbar-collapse" id="myNavbar">
            <li className="nav navbar-nav navbar-right">
              <li><a
                href=""
                className="dropdown-toggle"
                id="profile"
                data-toggle="dropdown"
                aria-haspopup="true"
                aria-expanded="true"
              >
                <span className="glyphicon glyphicon-user">
                  <span id="username"> {window.localStorage.getItem('username')}</span>
                  <span className="caret" />
                </span>
              </a>
                <ul className="dropdown-menu" aria-labelledby="profile">
                  <li><a href="#changepassword" data-toggle="modal" data-target="#changepassword">
                    <i className="glyphicon glyphicon-edit" /> Change password
                  </a>
                  </li>
                  <li role="separator" className="divider" />
                  <li><a href="/logout"><i className="glyphicon glyphicon-log-out" /> Logout</a></li>
                </ul>
              </li>
            </li>
          </div>
        </div>
      </nav>
    );
  }
}

export default Header;
