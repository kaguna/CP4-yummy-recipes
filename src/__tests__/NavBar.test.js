import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import NavBar from '../common/NavBar';

describe('Navbarcomponent', () => {
  const wrapper = shallow(<NavBar />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
