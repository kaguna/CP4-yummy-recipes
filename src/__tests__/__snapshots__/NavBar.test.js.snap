// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbarcomponent renders properly 1`] = `
<nav
  className="navbar navbar-default"
>
  <ChangePassword />
  <div
    className="container-fluid"
  >
    <div
      className="navbar-header"
    >
      <button
        className="navbar-toggle"
        data-target="#myNavbar"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
      <a
        className="navbar-brand"
        href=""
      >
        <i
          className="glyphicon glyphicon-cutlery"
        />
         Yummy Recipes
      </a>
    </div>
    <div
      className="collapse navbar-collapse"
      id="myNavbar"
    >
      <li
        className="nav navbar-nav navbar-right"
      >
        <ul>
          <li>
            <a
              aria-expanded="true"
              aria-haspopup="true"
              className="dropdown-toggle"
              data-toggle="dropdown"
              href=""
              id="profile"
            >
              <span
                className="glyphicon glyphicon-user"
              >
                <span
                  id="username"
                >
                   
                </span>
                <span
                  className="caret"
                />
              </span>
            </a>
            <ul
              aria-labelledby="profile"
              className="dropdown-menu"
            >
              <li>
                <a
                  data-target="#changepassword"
                  data-toggle="modal"
                  href="#changepassword"
                >
                  <i
                    className="glyphicon glyphicon-edit"
                  />
                   Change password
                </a>
              </li>
              <li
                className="divider"
                role="separator"
              />
              <li>
                <a
                  href="/logout"
                >
                  <i
                    className="glyphicon glyphicon-log-out"
                  />
                   Logout
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </div>
  </div>
</nav>
`;
